src/debugger/rnDebugSession.ts

```diff
@@ -116,6 +116,10 @@ export class RNDebugSession extends DebugSessionBase {
                             this.appLauncher.setAppWorker(this.appWorker);
 
                             this.appWorker.on("connected", (port: number) => {
+                                if (this.cancellationTokenSource.token.isCancellationRequested) {
+                                    return this.appWorker?.stop();
```

Comment:
Why we are trying to stop appworker here? What if the scenario will emergency stop? Will this mean that we don't receive any `connected` signals and we receive hanged appworker?